Source/WebCore/style/ElementRuleCollector.cpp

```diff
@@ -584,14 +584,65 @@ bool ElementRuleCollector::containerQueriesMatch(const RuleData& ruleData, const
 
 bool ElementRuleCollector::scopeRulesMatch(const RuleData& ruleData, const MatchRequest& matchRequest)
 {
-    auto queries = matchRequest.ruleSet.scopeRulesFor(ruleData);
+    auto rules = matchRequest.ruleSet.scopeRulesFor(ruleData);
 
-    if (queries.isEmpty())
+    if (rules.isEmpty())
         return true;
 
-    // FIXME: to implement
-    // https://bugs.webkit.org/show_bug.cgi?id=265241
-    return false;
+    SelectorChecker checker(element().rootElement()->document());
+    auto context = SelectorChecker::CheckingContext(SelectorChecker::Mode::ResolvingStyle);
```

Comment:
why is checker initialized using one syntax and the context other?

Reply:
No specific reason. It seems we don't have a coding style guideline for this.
Will use the non auto syntax for those.